api/cas-server-core-api-configuration/src/main/java/org/apereo/cas/configuration/model/support/jpa/AbstractJpaProperties.java

```diff
@@ -308,4 +339,24 @@ public long getFailFastTimeout() {
     public void setFailFastTimeout(final long failFastTimeout) {
         this.failFastTimeout = failFastTimeout;
     }
+
+    public void setType(final JpaType type) {
+        this.type = type;
+    }
+
+    public JpaType getType() {
+        return type;
+    }
+
+    public static enum JpaType {
+        /**
+         * RDBMS type using hibernate.
+         */
+        JDBC,
+        /**
+         * NoSQL backends using Kundera.
+         */
+        CASSANDRA
```

Comment:
This is likely not a good name for this behavior. Change to NOSQL or something like that.